sky/backends/backend_utils.py

```diff
@@ -1959,15 +1959,24 @@ def _update_cluster_status_no_lock(
         # Side effect: if the status is refreshed during autostopping, the
         # autostop field in the local cache will be reset, even though the
         # cluster will still be correctly stopped.
-        try:
-            backend = backends.CloudVmRayBackend()
-            backend.set_autostop(handle, -1, stream_logs=False)
-        except (Exception, SystemExit) as e:  # pylint: disable=broad-except
-            logger.debug(f'Failed to reset autostop. Due to '
-                         f'{common_utils.format_exception(e)}')
-        global_user_state.set_cluster_autostop_value(handle.cluster_name,
-                                                     -1,
-                                                     to_down=False)
+        backend = get_backend_from_handle(handle)
+        if isinstance(backend,
+                      backends.CloudVmRayBackend) and record['autostop'] >= 0:
+            if not backend.is_autostopping(handle, stream_logs=False):
+                try:
+                    backend.set_autostop(handle, -1, stream_logs=False)
+                except (Exception, SystemExit) as e:  # pylint: disable=broad-except
+                    logger.debug(f'Failed to reset autostop. Due to '
+                                 f'{common_utils.format_exception(e)}')
+                global_user_state.set_cluster_autostop_value(
+                    handle.cluster_name, -1, to_down=False)
+            else:
+                ux_utils.console_newline()
+                operation_str = 'autodowning' if record[
+                    'to_down'] else 'autostopping'
+                logger.info(
+                    f'Cluster {cluster_name!r} is {operation_str}, it will remain '
+                    'in INIT state until the autostop is finished.')
```

Comment:
```suggestion
                    f'Cluster {cluster_name!r} is {operation_str}. Setting to INIT status; try refresh again in a while.')
```
(to hint that you need to refresh to get rid of INIT)